bpf/bpf_host.c

```diff
@@ -880,6 +880,9 @@ tail_handle_ipv4(struct __ctx_buff *ctx, __u32 ipcache_srcid, const bool from_ho
 	int ret;
 	__s8 ext_err = 0;
 
+	if (src_sec_identity == 0)
+		src_sec_identity = WORLD_IPV4_ID;
+
```

Comment:
Imho this change (and the corresponding one in bpf_overlay) is way too broad, and can have unexpected side-effects for existing code. Why not push this "override" logic down into `nodeport_lb4()` ?

Reply:
see my previous comment, @brb asked offline to have the caller of `nodeport_lb4()` set the src identity so I just followed that, but I don't mind moving it a bit deeper in the call stack

Reply:
hm yes, but you're changing the calling paths at a much higher level - it's not obvious that **all** subsequent users of the src_sec_identity want to have this override. What if one of them has unique handling for `identity == 0`, that now doesn't get applied any more?

I think you'll have a much easier life keeping this contained inside the nodeport-scope for now.

Reply:
> hm yes, but you're changing the calling paths at a much higher level

yes, on purpose in order to have a consistent value across all users rather than having it set to 0 in one code path and WORLD_IPV4_ID in another

> it's not obvious that all subsequent users of the src_sec_identity want to have this override. What if one of them has unique handling for identity == 0, that now doesn't get applied any more?

fair, perhaps I have too much faith in the tests :disappointed: 

> I think you'll have a much easier life keeping this contained inside the nodeport-scope for now.

sounds good, I'll go with that then :ok_hand:

Reply:
Just checking - this isn't updated in the latest push, right? The identity override is still in every calling path, not down in `nodeport_lb4()`.

Reply:
ah, I understood we just wanted to move it right before calling into `nodeport_lb4()` and not in that function, let me move it back down to that function then